dao/src/main/java/org/thingsboard/server/dao/device/DeviceConnectivityServiceImpl.java

```diff
@@ -115,15 +128,50 @@ public JsonNode findDevicePublishTelemetryCommands(String baseUrl, Device device
 
     @Override
     public Resource getPemCertFile(String protocol) {
-        String certFilePath = deviceConnectivityConfiguration.getConnectivity()
-                .get(protocol)
-                .getPemCertFile();
+        return certs.computeIfAbsent(protocol, key -> {
```

Comment:
I would like to see the error log message if the respective cert is not found. e.g., an Application-ready event with the lowest priority or something similar